### ISSUE TITLE
Production Server Setup with Docker : lockfile error

### ISSUE THREAD

**Issue by chbon** [4minitz/4minitz#465]
Hi,

I have been following the admin guide to install 4Minitz with Docker.

After executing `docker run -it --rm --volumes-from 4minitz-storage -p 3100:3333 4minitz/4minitz` the installation is OK and accessible, if restarting docker i've got the following error :

> Inside container basedir: /4minitz_storage - OK
> *** ERROR ***
> Found lockfile /4minitz_storage/4minitz.lock.
> Another container with 4Minitz seems to be using
> the same host storage directory. Please first execute:
> docker stop 0e165a6102ed
>

The ID 0e165a6102ed does not exist and does not corresponds to any containers.

Thanks for your help.

**Comment by derwok** [comment]
Thanks for the feedback.
TL;DR
If the lock file is there and you are sure no 4Minitz container is running, you may sefely just delete the lock file and spin up your new container. 

The lock file has "no magic" it is just a safety net to prohibit that two containers store their stuff into the same directory hierarchy. The 4minitz.kock file is generated in our 4Minitz docker launch script `.docker/4minitz.sh` right at the begining. It is then deleted at the very end of the script. 

For a few years now we did not have any issues with an orphan lock file. 

If your container is not running anymore and the lock file is there - could you answer a few questions:
- What platform (OS) is your docker running on? Which version?
- What exact version of docker are you using?
- What version of 4Minitz? current stable? 1.5.2?
- How exactly did you terminate your 4Minitz container?

Thanks for your help.

**Comment by chbon** [comment]
Hello,

Thanks for your explanations, here is some details :

- What platform (OS) is your docker running on? Which version?
Windows 10 (1607)
- What exact version of docker are you using?
Docker  18.06.1-ce
- What version of 4Minitz? current stable? 1.5.2?
Current stable yes
- How exactly did you terminate your 4Minitz container?
Docker service restarted or computer restarted (no problem with Ctr+C or docker stop)

**Comment by RedYouMan** [comment]
I had same problem but i before to do anything with docker i must do Cntrl-C of processes. 
After i have restarted docker i can do 
docker run -it --rm --volumes-from 4minitz-storage -p 3100:3333  4minitz/4minitz

**Comment by derwok** [comment]
The lock file is a means to prevent an admin to accidently start two docker instances that try to store data into the same directory. This will probably lead to data corruption and so we try to avoid it as good as possible.

Nevertheless, I could reproduce your issue only when I "Hard Stop the Docker Server". Please don't do this. Currently the intended way is to "CTRL+C" end the 4Minitz docker container. Then restart docker server / docker demon. This also ensures that no data is lost that the database still has to flush to the disk. It's like pulling the power cord of your desktop PC (without having a battery). You should not do this.

If you know what you are doing, and you know you don't have two instances of 4Minitz running, you may safely delete the lock file.

**Comment by Loofay** [comment]
Hi All,

my server crashed because of a powercut and I had the same issue. The Container didn't start until I delete the /4minitz_storage/4minitz.lock.

Now, the Server is Up n Running, but I can't login with any account.

May I ask you for an Idear to solve this?

Thanks for your help.

**Comment by mraninja** [comment]
I have this same issue:

> After executing `docker run -it --rm --volumes-from 4minitz-storage -p 3100:3333 4minitz/4minitz` the installation is OK and accessible, if restarting docker i've got the following error :
> 
> > Inside container basedir: /4minitz_storage - OK
> > *** ERROR ***
> > Found lockfile /4minitz_storage/4minitz.lock.
> > Another container with 4Minitz seems to be using
> > the same host storage directory. Please first execute:
> > docker stop 0e165a6102ed
> 
> The ID 0e165a6102ed does not exist and does not corresponds to any containers."


Tried this:

"I had same problem but i before to do anything with docker i must do Cntrl-C of processes.
After i have restarted docker i can do
docker run -it --rm --volumes-from 4minitz-storage -p 3100:3333 4minitz/4minitz"

No luck with this.

I was able to get 4Minitz! up and running with Win 10 WSL, Docker, mongoDB, Ubuntu 20.04.1 LTS (Focal Fossa).

--Thank you
